common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -341,25 +348,45 @@ def _render_student_view_for_items(self, context, display_items, fragment):
         display_items.  Returns a list of dict objects with information about
         the given display_items.
         """
-        bookmarks_service = self.runtime.service(self, "bookmarks")
-        context["username"] = self.runtime.service(self, "user").get_current_user().opt_attrs['edx-platform.username']
+        is_user_authenticated = context.get('user_authenticated', False)
+        bookmarks_service = self.runtime.service(self, 'bookmarks')
+        context['username'] = self.runtime.service(self, 'user').get_current_user().opt_attrs.get(
+            'edx-platform.username')
         display_names = [
             self.get_parent().display_name_with_default,
             self.display_name_with_default
         ]
         contents = []
         for item in display_items:
-            is_bookmarked = bookmarks_service.is_bookmarked(usage_key=item.scope_ids.usage_id)
-            context["bookmarked"] = is_bookmarked
+            # NOTE (CCB): This seems like a hack, but I don't see a better method of determining the type/category.
+            item_type = item.get_icon_class()
+            usage_id = item.scope_ids.usage_id
+
```

Comment:
I don't have a console open, but if `item` is an XModule or XModuleDescriptor, I think what you want is:

```python
usage_key = item.location
block_type = usage_key.block_type
```